crates/bevy_input/src/gamepad.rs

```diff
@@ -2,6 +2,53 @@ use crate::{Axis, Input};
 use bevy_ecs::event::{EventReader, EventWriter};
 use bevy_ecs::system::{Res, ResMut, Resource};
 use bevy_utils::{tracing::info, HashMap, HashSet};
+use thiserror::Error;
+
+/// Errors that occur when setting axis settings for gamepad input.
+///
+/// + `ZoneBounds`
+///
+///     The given parameters must fit the bounds described by the expression
+///     -1.0 <= `livezone_lowerbound` <= `deadzone_lowerbound` <= 0.0 <= `deadzone_upperbound` <= `livezone_upperbound` <= 1.0.
+///     These are named parameters to `AxisSettings::new()`.
+///
+/// + `Threshold`
+///
+///     The given parameter was not in range 0.0..=2.0.
+#[derive(Error, Debug)]
+pub enum AxisSettingsError {
+    #[error("the conditions -1.0 <= livezone_lowerbound <= deadzone_lowerbound <= 0.0 <= deadzone_upperbound <= livezone_upperbound <= 1.0 must hold true")]
+    ZoneBounds,
```

Comment:
I think this returned error will be more useful if you add named fields to this variant for the four values.

Reply:
(painfully) Done.